src/server/pfs/server/url_worker.go

```diff
@@ -99,20 +112,41 @@ func (d *driver) processGetFileURLTask(ctx context.Context, task *GetFileURLTask
 	if err != nil {
 		return nil, errors.Wrapf(err, "error parsing URL %v", task.URL)
 	}
-	objClient, err := obj.NewClientFromURLAndSecret(ctx, url, false)
+	bucket, err := openBucket(ctx, url)
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
+		if err := bucket.Close(); err != nil {
+			retErr = multierror.Append(retErr, errors.Wrapf(err, "error closing bucket %s", url.Bucket))
+		}
+	}()
 	if err := log.LogStep(ctx, "getFileURLTask", func(ctx context.Context) error {
 		err := src.Iterate(ctx, func(fi *pfs.FileInfo, file fileset.File) error {
 			if fi.FileType != pfs.FileType_FILE {
 				return nil
 			}
-			return miscutil.WithPipe(func(w io.Writer) error {
-				return errors.EnsureStack(file.Content(ctx, w))
-			}, func(r io.Reader) error {
-				return errors.EnsureStack(objClient.Put(ctx, filepath.Join(url.Object, fi.File.Path), r))
-			})
+			exists, err := bucket.Exists(ctx, fi.File.Path)
```

Comment:
I don't think we want this check. Object storage puts are idempotent and this prevents us from overwriting a file in a subsequent get file url. We may want to consider a more proper sync later though, because we won't be deleting files with either approach.